### PR TITLE
Support for changing datasets and code differencing tools by options

### PR DESCRIPTION
**Pull request by young170**
Changed files summary:
* components/LCE/*.csv
    * add GBR vector data
    * format "vector_file_{dataset}\_{differencing}.csv" and "commit_file_{dataset}.csv"
* core/ChangeCollector/app/src/main/java/ChangeCollector/ChangeVector.java
    * change vectors for LAS
* core/LCE/app/src/main/java/LCE/ChangeVector.java
    * change vectors for LAS
* core/ChangeCollector/app/src/main/java/ChangeCollector/Extractor.java
    * add support for LAS
* core/LCE/app/src/main/java/LCE/App.java
    * add support for GBR
    * currently GBR has conflicts with D4J so works by directly copying the stored files
* core/LCE/app/src/main/java/LCE/LCS.java
    * fix LCS algorithm
* core/LCE/app/src/main/java/LCE/GitLoader.java
    * identical fix to `f09f0bd0b75536346e8571898ff70a9a4147eff4`
* core/LCE/app/src/main/java/LCE/Extractor.java
    * identical fix to `a06fea6f5b7c4dd1e8df33516ad4e92688cf6fd8`
* core/LCE/app/src/main/java/LCE/CosineSimilarity.java
    * add support for applying cosine text similarity on the code differencing results
* core/LCE/app/src/main/java/LCE/App.java
    * add support for applying cosine text similarity on the code differencing results

### CODE REVIEW

File: launcher_v2.py [hunk 1]
```diff
@@ -27,10 +27,17 @@ def parse_argv() -> tuple:
     parser.add_argument("-r", "--rebuild",  action = "store_true",
                         help = "Rebuilds all SPI submodules if enabled.")
     
-    parser.add_argument("-t","--textsim", action = "store_true", help = "On and Off options to use text sim or not")
+    parser.add_argument("-t","--textsim", type = str, default = "false",
+                        help = "Specifies target of text sim (diff, tree)")
```
Comment (newwin01) [inline launcher_v2.py:31]: Please add more detailed info about text sim methods

File: launcher_v2.py [hunk 2]
```diff
@@ -27,10 +27,17 @@ def parse_argv() -> tuple:
...
 
     parser.add_argument("-A", "--APR",   type = str,     default = "ConFix",
                         help = "Tells on what APR do you want to use.")
+
+    parser.add_argument("-D", "--Differencing",   type = str,     default = "GumTree4.0",
+                        help = "Specifies code differencing tool to be used (GumTree3.0, GumTree4.0, and LAS).")
```
Comment (newwin01) [inline launcher_v2.py:37]: Need to find a way to switch gumtree version on build.gradle. Using Jar might be the possible solution.

File: core/ChangeCollector/app/src/main/java/ChangeCollector/Extractor.java [hunk 3]
```diff
@@ -149,24 +153,45 @@ public boolean extract_gumtree_log(String repo_path, String diff_path, String ou
...
+                if (code_differencing_tool.equals("LAS")) {
+                    File BBICFile = new File(output_dir + "/" + repo_name.substring(0, 1).toUpperCase() + repo_name.substring(1), "BBIC.java");
+                    File BICFile = new File(output_dir + "/" + repo_name.substring(0, 1).toUpperCase() + repo_name.substring(1), "BIC.java");
 
-                Matcher defaultMatcher = Matchers.getInstance().getMatcher();
-                MappingStore mappings = defaultMatcher.match(src, dst);
-                EditScriptGenerator editScriptGenerator = new SimplifiedChawatheScriptGenerator();
-                EditScript actions = editScriptGenerator.computeActions(mappings);
+                    ProcessBuilder processBuilder = new ProcessBuilder(
+                        "java", "-cp", project_root+"/las-1.0.0-SNAPSHOT-jar-with-dependencies.jar", "main.LAS", BBICFile.getPath(), BICFile.getPath()
```
Comment (newwin01) [inline core/ChangeCollector/app/src/main/java/ChangeCollector/Extractor.java:167]: Please add Jar file for LAS

**Review by newwin01 — CHANGES_REQUESTED** [review]
Please see the comments and update README.md as well. 

I appreciate your effort!

File: launcher_v2.py (hunk 2 above)
Comment (young170) [reply launcher_v2.py:37]: I will do this later...
image

**Review by newwin01 — APPROVED** [review]
LGTM

**Pull request merged**